### PR TITLE
Transformer roles

### PR DESCRIPTION
**Pull request by JFranklin3000**
Closes [issue URL]

##### In this PR
- Added missing Items from Transformers and FGtAA

##### Testing
- 
fvtt package pack -n "dditems"
fvtt package pack -n "eocitems"
fvtt package pack -n "tfcrbitems"
fvtt package pack -n "fgtaaitems"

### CODE REVIEW

File: module/sheet-handlers/role-handler.mjs [hunk 1]
```diff
@@ -319,14 +372,24 @@ export class RoleHandler {
...
       const essenceString = `system.essenceLevels.${essence}`;
       const essenceRankString = `system.essenceRanks.${essence}`;
-      const rankValue = CONFIG.E20.MLPAdvancement[rank];
+      let rankValue = [];
+      if (role.system.version == "transformers"){
+        rankValue = CONFIG.E20.TFSpecialAdvancement[rank];
+        if (level1Essences[essence]) {
+          rankValue.push("level1");
+        }
+
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:388]:
```suggestion
```

File: module/sheet-handlers/role-handler.mjs [hunk 2]
```diff
@@ -319,14 +372,24 @@ export class RoleHandler {
    * @param {Role} role The role that was dropped on the actor
    * @param {Function} dropFunc The function for the drop of the role
    */
-  async _setEssenceProgression (options, role, dropFunc) {
+  async _setEssenceProgression (options, role, dropFunc, level1Essences) {
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:375]:
```suggestion
  async _setEssenceProgression(options, role, dropFunc, level1Essences) {
```

File: module/sheet-handlers/role-handler.mjs [hunk 3]
```diff
@@ -191,7 +191,9 @@ export class RoleHandler {
 
     if (role.system.version == 'myLittlePony') {
       await this._selectEssenceProgression(role,dropFunc);
-    } else {
+    } else if (role.system.hasSpecialAdvancement) {
+      await this._selectFirstEssences(role,dropFunc);
+    }else{
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:196]:
```suggestion
    } else {
```

File: module/sheet-handlers/role-handler.mjs [hunk 4]
```diff
@@ -259,15 +261,66 @@ export class RoleHandler {
...
+  _verifySelection(options) {
+    let selectionAmount = 0;
+    for (const [, selection] of Object.entries(options)) {
+      if (selection == true) {
+        selectionAmount += 1;
+      }
+    }
+
+    if (selectionAmount != 2) {
+      throw new Error('Must Select 2 Essences');
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:303]: Localize

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/sheet-handlers/role-handler.mjs [hunk 5]
```diff
@@ -259,15 +261,66 @@ export class RoleHandler {
     this._actor.setFlag('essence20', 'previousLevel', 0);
   }
 
+
+  async _selectFirstEssences(role,dropFunc) {
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:265]:
```suggestion
  async _selectFirstEssences(role, dropFunc) {
```

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/sheet-handlers/role-handler.mjs [hunk 6]
```diff
@@ -261,7 +261,11 @@ export class RoleHandler {
     this._actor.setFlag('essence20', 'previousLevel', 0);
   }
 
-
+  /**
+  *
+  * @param {Object} role The role that is being dropped on the actor
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:266]:
```suggestion
  * @param {Role} role The role that is being dropped on the actor
```

File: module/sheet-handlers/role-handler.mjs [hunk 7]
```diff
@@ -291,6 +295,11 @@ export class RoleHandler {
 
   }
 
+  /**
+   *
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:299]:
```suggestion
```

File: module/sheet-handlers/role-handler.mjs [hunk 8]
```diff
@@ -291,6 +295,11 @@ export class RoleHandler {
...
+   * @param {Object} options The options selected in the previous dialog
+   * @returns
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:301]:
```suggestion
```

File: lang/en.json [hunk 9]
```diff
@@ -242,6 +242,7 @@
     "EssenceRankSilver": "Silver",
     "EssenceRankThird": "Third",
     "EssenceProgressionSelect": "Select Essence Progression",
+    "EssencesRequiredError": "Must Select 2 Essences",
```
Comment (phildominguez) [inline lang/en.json:245]:
```suggestion
    "EssencesRequiredError": "Must select 2 Essences",
```

**Review by phildominguez — CHANGES_REQUESTED** [review]

**Review by phildominguez — APPROVED** [review]

File: module/sheet-handlers/role-handler.mjs (hunk 4 above)
Comment (JFranklin3000) [reply module/sheet-handlers/role-handler.mjs]: fixed

**Pull request merged**